pkg/controller/oneagent/istio.go

```diff
@@ -17,105 +17,152 @@ import (
+	if err != nil {
+		return false, err
+	}
+
+	seUpd, err := r.reconcileIstioRemoveConfiguration(instance, istio.ServiceEntryGVK, listOps, seen, logger)
+	if err != nil {
+		return false, err
+	}
+
+	return vsUpd || seUpd, nil
 }
 
 func (r *ReconcileOneAgent) reconcileIstioRemoveConfiguration(instance *dynatracev1alpha1.OneAgent, gvk schema.GroupVersionKind,
-	listOps *client.ListOptions, seen map[string]bool, logger logr.Logger) {
+	listOps *client.ListOptions, seen map[string]bool, logger logr.Logger) (bool, error) {
 
 	var list unstructured.UnstructuredList
 	list.SetGroupVersionKind(gvk)
 
 	if err := r.client.List(context.TODO(), listOps, &list); err != nil {
-		return
+		return false, err
 	}
 
+	upd := false
+
 	for _, item := range list.Items {
 		if _, ok := seen[item.GetName()]; !ok {
+			upd = true
 			logger.Info(fmt.Sprintf("removing Istio %s: %v", gvk.Kind, item.GetName()))
 			if err := r.client.Delete(context.TODO(), &item); err != nil {
-				logger.Info(fmt.Sprintf("failed to delete Istio %s: %v", gvk.Kind, err))
-				continue
+				return false, fmt.Errorf("failed to delete Istio %s: %v", gvk.Kind, err)
 			}
 		}
 	}
+
+	return upd, nil
 }
 
 func (r *ReconcileOneAgent) reconcileIstioCreateConfigurations(instance *dynatracev1alpha1.OneAgent,
-	comHosts []dtclient.CommunicationHost, logger logr.Logger) error {
+	comHosts []dtclient.CommunicationHost, role string, logger logr.Logger) (bool, error) {
+
+	upd := false
 
 	for _, ch := range comHosts {
 		name := istio.BuildNameForEndpoint(instance.Name, ch.Host, ch.Port)
 
 		if notFound := r.configurationExists(istio.ServiceEntryGVK, instance.Namespace, name); notFound {
 			logger.Info("creating Istio ServiceEntry", "objectName", name, "host", ch.Host, "port", ch.Port)
 			payload := istio.BuildServiceEntry(name, ch.Host, ch.Port, ch.Protocol)
-			if err := r.reconcileIstioCreateConfiguration(instance, istio.ServiceEntryGVK, payload); err != nil {
-				logger.Info(fmt.Sprintf("failed to create Istio ServiceEntry: %v", err))
+			if err := r.reconcileIstioCreateConfiguration(instance, istio.ServiceEntryGVK, role, payload); err != nil {
+				return false, fmt.Errorf("failed to create Istio ServiceEntry: %v", err)
 			}
+			upd = true
 		}
 
 		if notFound := r.configurationExists(istio.VirtualServiceGVK, instance.Namespace, name); notFound {
 			logger.Info("creating Istio VirtualService", "objectName", name, "host", ch.Host, "port", ch.Port, "protocol", ch.Protocol)
 			payload := istio.BuildVirtualService(name, ch.Host, ch.Port, ch.Protocol)
-			if err := r.reconcileIstioCreateConfiguration(instance, istio.VirtualServiceGVK, payload); err != nil {
-				logger.Info(fmt.Sprintf("failed to create Istio VirtualService: %v", err))
+			if err := r.reconcileIstioCreateConfiguration(instance, istio.VirtualServiceGVK, role, payload); err != nil {
+				return false, fmt.Errorf("failed to create Istio VirtualService: %v", err)
```

Comment:
Is exiting the function here the right choice?

Logging the failure and continue could be also ok-ish.